activerecord/lib/active_record/connection_handling.rb

```diff
@@ -192,14 +185,13 @@ def connection
       retrieve_connection
     end
 
-    attr_writer :connection_specification_name
+    attr_writer :connection_handler
```

Comment:
Can we instead just have the model specify the schema name?

```ruby
  class_attribute :schema_name

  alias_method :connection_specification_name, :schema_name
  alias_method :connection_specification_name=, :schema_name=
  deprecate :connection_specification_name, connection_specification_name=
```

then detach the connection handling from the model and just look it up from the model using the `schema_name`.  That way we can do things like switch the connection for a schema as part of shard migrations and the connection switch affect the sharded models.